core/util.cpp

```diff
@@ -167,7 +169,9 @@ static IconDatabase readIconData()
         const QDir dir(fileInfo.path());
         const QString className = dir.dirName();
         const QString iconName = fileInfo.fileName();
-        IconCacheEntry &perClassData = data[className.toLatin1()];
+        const QByteArray classNameAsByteArray = className.toLatin1();
+        IconCacheEntry &perClassData = data[QLatin1String(classNameAsByteArray)];
```

Comment:
You might want to pass `data()` + `size()` here, as the `QLatin1String(QByteArray)` ctor performs a (useless) `strlen()`.